salt/modules/win_iis.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+'''
+Microsoft IIS site management via WebAdministration powershell module
+
+:platform:      Windows
+
+'''
+
+
+from __future__ import absolute_import
+
+# Import salt libs
+import salt.utils
+
+# Define the module's virtual name
+__virtualname__ = 'win_iis'
+
+
+def __virtual__():
+    '''
+    Load only on Windows
+    '''
+    if salt.utils.is_windows():
+        return __virtualname__
+    return False
+
+
+def _srvmgr(func):
+    '''
+    Execute a function from the WebAdministration PS module
+    '''
+
+    return __salt__['cmd.run'](
+        'Import-Module WebAdministration; {0}'.format(func),
+        shell='powershell',
+        python_shell=True)
+
+
+def list_sites():
+    '''
+    List all the currently deployed websites
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' win_iis.list_sites
+    '''
+    pscmd = []
+    pscmd.append(r"Get-WebSite -erroraction silentlycontinue -warningaction silentlycontinue")
+    pscmd.append(r" | foreach {")
+    pscmd.append(r" $_.Name")
+    pscmd.append(r"};")
+
+    command = ''.join(pscmd)
+    return _srvmgr(command)
+
+
+def create_site(
+        name,
+        protocol,
+        sourcepath,
+        port,
+        apppool='',
+        hostheader='',
+        ipaddress=''):
+    '''
+    Create a basic website in IIS
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' win_iis.create_site name='My Test Site' protocol='http' sourcepath='c:\\stage' port='80' apppool='TestPool'
+
+    '''
+
+    pscmd = []
+    pscmd.append(r"cd IIS:\Sites\;")
```

Comment:
We prefer single-quotes versus double-quotes. Please see our style guide. https://docs.saltstack.com/en/latest/topics/development/conventions/style.html